### PR TITLE
Docs:  use YAML instead of Diff in docker-compose.yml

### PR DESCRIPTION
**Pull request by ghost**
- Using `yaml` in the document will be more practical than `diff`
- Add some comments in YAML
- Fix docker-compose.yml in Chinese translations

### CODE REVIEW

File: docs/content/doc/installation/with-docker.en-us.md [hunk 1]
```diff
@@ -225,8 +223,7 @@ services:
     networks:
       - gitea
     volumes:
--     - ./gitea:/data
-+     - gitea:/data
+      - ./gitea:/data
```
Comment (lafriks) [inline docs/content/doc/installation/with-docker.en-us.md]: If there is volume defined than this should be `- gitea:/data`

**Review by lafriks — CHANGES_REQUESTED** [review]

**Comment by KN4CK3R** [comment]
The diff was intended to show what should be changed in contrast to the default. Now you have to compare both to see what should be changed.

**Comment by penlinux** [comment]
> The diff was intended to show what should be changed in contrast to the default. Now you have to compare both to see what should be changed.

Yes, this highlights what has been changed

but I think it is docker's job to teach users how to use docker compsoe. We should directly provide the standard `docker-compose.yml`, which is easy to use

File: docs/content/doc/installation/with-docker.en-us.md (hunk 1 above)
Comment (ghost) [reply docs/content/doc/installation/with-docker.en-us.md]: Done.

**Comment by Gusted** [comment]
> but I think it is docker's job to teach users how to use docker compsoe. 

However they will complain to Gitea instead to Docker if they fail to understand(or mess up) how to setup certain configurations. There's a shared responsibility IMO.

**Comment by penlinux** [comment]
> However they will complain to Gitea instead to Docker if they fail to understand(or mess up) how to setup certain configurations. There's a shared responsibility IMO.

In fact, users just copy-paste the configuration they need from the docs, and diff has caused some complaints. Now, I've added some comments in the YAML to help people understand .

**Pull request closed without merging**